addons/mass_mailing/tests/test_mailing_ab_testing.py

```diff
@@ -147,3 +147,21 @@ def test_ab_testing_mail_compare_version(self):
         # Record with mailing_type equals to mail should be included in compare version count, doesn't matter
         # if campign_id is same
         self.assertEqual(compare_version_record_count, 2, 'Number of records should be according to mailing type')
+
+    @mute_logger('odoo.addons.mail.models.mail_mail')
+    @users('user_marketing')
+    def test_send_minimum_one_mail_for_ab_testing(self):
+        """ Test that it should send minimum one mail(if possible) when ab_testing_pc is zero."""
+        ab_testing = self.env['mailing.mailing'].create({
+            'subject': 'A/B Testing SMS V1',
+            'contact_list_ids': self.mailing_list.ids,
```

Comment:
Since "mailing_list" contains 150 records, it makes it complicated to test your use case.

You should probably make it so "2%" is less that one participant.
So maybe something like:

```py
mailing_list = self._create_mailing_list_of_x_contacts(10)
...
    'contact_list_ids': mailing_list.ids,
...
```

And explain in the method docstring.
"We have 10 persons in the contact list and want to contact only 2% of them, which would result to "0.2 person", let's make sure we at least send one email."

Do you see what I mean?